File: Backend/YouBook/src/main/java/com/youbook/YouBook/entities/Role.java
package com.youbook.YouBook.entities;

import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.persistence.Entity;
import java.io.Serializable;
@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class Role implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @NotNull
    private String name;
}

File: youbooking-frontEnd/src/app/models/role.ts
export class Role {
  private _id!:BigInt;
  private _name!:String;

  constructor(name: String) {
    this._name = name;
  }

  get id(): BigInt {
    return this._id;
  }

  get name(): String {
    return this._name;
  }

  set name(value: String) {
    this._name = value;
  }
}
